src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,39 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  #this;
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+  hide() {
```

Comment:
The 'hide' method correctly sets the 'hidden' property to true without toggling it, following the task checklist requirement for function naming.